tests/test_order.py

```diff
@@ -1,106 +1,112 @@
+        requests_mock.get(url=ORDER_URL, json=info)
+        order_obj = order.Order(auth_mock, order_id=constants.ORDER_ID, order_info=info)
+        assert order_obj.order_details == expected
+
+    @pytest.mark.parametrize(
+        "info, expected",
+        [
+            ({"status": "FULFILLED"}, True),
+            ({"status": "OTHER STATUS"}, False),
+        ],
+        ids=["FULFILLED", "OTHER STATUS"],
     )
-    assert order_mock.order_details == order_details
-
-
-@pytest.mark.parametrize(
-    "status,expected",
-    [("NOT STARTED", False), ("PLACED", False), ("FULFILLED", True)],
-)
-def test_is_fulfilled(order_mock, status, expected, monkeypatch):
-    monkeypatch.setattr(order.Order, "info", {"status": status})
-    assert order_mock.is_fulfilled == expected
-
-
-def test_order_parameters(order_mock):
-    assert not order_mock.order_parameters
-
-
-def test_get_assets_should_search_assets_by_order_id(auth_mock, requests_mock):
-    order_response = {"id": constants.ORDER_ID, "status": "FULFILLED"}
-
-    url_order_info = f"{constants.API_HOST}/v2/orders/{constants.ORDER_ID}"
-    requests_mock.get(url=url_order_info, json=order_response)
-    url_asset_info = f"{constants.API_HOST}/v2/assets?search={constants.ORDER_ID}&size=50"
-    requests_mock.get(url=url_asset_info, json=fixtures_order.JSON_GET_ASSETS_RESPONSE)
-    order_placed = order.Order(auth=auth_mock, order_id=constants.ORDER_ID)
-    (asset_returned,) = order_placed.get_assets()
-    assert isinstance(asset_returned, asset.Asset)
-    assert asset_returned.asset_id == constants.ASSET_ORDER_ID
-
-
-@pytest.mark.parametrize(
-    "status",
-    [
-        "CREATED",
-        "BEING_PLACED",
-        "PLACED",
-        "PLACEMENT_FAILED",
-        "DELIVERY_INITIALIZATION_FAILED",
-        "BEING_FULFILLED",
-        "DOWNLOAD_FAILED",
-        "DOWNLOADED",
-        "FAILED_PERMANENTLY",
-    ],
-)
-def test_should_fail_to_get_assets_for_unfulfilled_order(auth_mock, requests_mock, status):
-    order_response = {"id": constants.ORDER_ID, "status": status}
-    url_order_info = f"{constants.API_HOST}/v2/orders/{constants.ORDER_ID}"
-    requests_mock.get(url=url_order_info, json=order_response)
-    order_placed = order.Order(auth=auth_mock, order_id=constants.ORDER_ID)
-    with pytest.raises(ValueError):
-        order_placed.get_assets()
+    def test_should_compute_is_fulfilled(self, auth_mock, requests_mock: req_mock.Mocker, info: dict, expected):
+        requests_mock.get(url=ORDER_URL, json=info)
+        order_obj = order.Order(auth_mock, order_id=constants.ORDER_ID, order_info=info)
+        assert order_obj.is_fulfilled == expected
+
+    def test_get_assets_should_search_assets_by_order_id(self, auth_mock, requests_mock):
+        order_response = {"id": constants.ORDER_ID, "status": "FULFILLED"}
+        requests_mock.get(url=ORDER_URL, json=order_response)
```

Comment:
We don't need to mock this if we're passing it in the constructor

Reply:
We need to pass auth_mock for calling the get_assets method, and if so it will try to call the info endpoint as well. For clarity I will not pass info in the order initialization.